detectors/aws/eks/detector.go

```diff
@@ -139,50 +139,34 @@ func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
 	return err == nil && !info.IsDir()
 }
 
-// fetchString executes an HTTP request with a given HTTP Method and URL string.
-func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
-	request, err := http.NewRequest(httpMethod, URL, nil)
-	if err != nil {
-		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
-	}
-
-	// Set HTTP request header with authentication credentials
-	authHeader, err := getK8sCredHeader()
+// fetchString executes an HTTP Kubernetes client request with a given HTTP Method and URL string.
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string, API string) (string, error) {
```

Comment:
I'm not sure that this is the appropriate interface, it feels too low-level and I'd definitely like to avoid creating a k8s clientset for every invocation.  Since the k8s client seems to be used to retrieve two config maps, perhaps exposing a forwarding method for `GetConfigMap(namespace, name)` or the like would be better.